File: GeeksforGeeks/Values_with_Equal_Array_Remainders.py
import math


class Solution:

  def sameMod(self, arr: list[int]) -> int:
    n = len(arr)
    g = 0

    # Find the GCD of differences from the first element
    for i in range(1, n):
      g = math.gcd(g, abs(arr[i] - arr[0]))

    # If all elements are equal, 'g' remains 0 (infinitely many valid k values)
    if g == 0:
      return -1

    # Count the positive divisors of g
    count = 0
    i = 1
    while i * i <= g:
      if g % i == 0:
        count += 1

        # If i and g // i are distinct divisors, count both
        if i != g // i:
          count += 1
      i += 1

    return count

File: GeeksforGeeks/Values_with_Equal_Array_Remainders.cpp
class Solution {
  public:
    int sameMod(vector<int> &arr) {
        // code here
        int n = arr.size();

        int g = 0;

        // Find GCD of differences
        for (int i = 1; i < n; i++) {
            g = gcd(g, abs(arr[i] - arr[0]));
        }

        // All elements are equal
        if (g == 0)
            return -1;

        // Count divisors of g
        int count = 0;

        for (int i = 1; i * i <= g; i++) {
            if (g % i == 0) {
                count++;

                // i and g/i are two different divisors
                if (i != g / i)
                    count++;
            }
        }

        return count;
    }
};

